comm-service/controller/comm-controller.js

```diff
@@ -0,0 +1,63 @@
+import { sendJoinRoomFail } from '../services/socket.js'
+
+
+var roomInterviewers = {};
+const interviewerSwitchEvent = 'interviewer switch event';
+const interviewerSwitchRequestEvent = 'request interviewer switch';
+
+export function registerChatHandlers(io, clientSocket) {
+    clientSocket.on('join room', eventData => {
+        if (!eventData.room_id) {
+            console.log("Join room event emitted without room_id");
+            sendJoinRoomFail(clientSocket, { message: "missing room_id" });
+        }
+        const room_id = eventData.room_id;
+        const username = eventData.username;
```

Comment:
In case you didn't know, we are able to get the username directly from the socket.

Reply:
Yep thank you! For now i shall not change the implementation, but can do so in the future